river/Cursor.zig

```diff
@@ -1049,21 +1050,33 @@ pub fn checkFocusFollowsCursor(self: *Self) void {
     if (self.seat.drag == .pointer) return;
     if (server.config.focus_follows_cursor == .disabled) return;
     if (self.surfaceAt()) |result| {
-        if (server.config.focus_follows_cursor == .always or
-            self.seat.wlr_seat.pointer_state.focused_surface != result.surface)
-        {
-            switch (result.parent) {
-                .view => |view| {
-                    if (self.seat.focused != .view or self.seat.focused.view != view) {
-                        self.seat.focusOutput(view.output);
-                        self.seat.focus(view);
-                        server.root.startTransaction();
-                    }
-                },
-                .layer_surface, .lock_surface => {},
-                .xwayland_override_redirect => assert(build_options.xwayland),
-            }
+        switch (result.parent) {
+            .view => |view| {
+                // Don't re-focus the last focused view when the mode is .normal
+                if (server.config.focus_follows_cursor == .normal and
+                    view == self.last_followed_view) return;
+                // Some windows have a input region bigger than their window
+                // geometry, we only want to move focus when the cursor
+                // properly enters the window (the box that we draw borders around)
+                var output_layout_box: wlr.Box = undefined;
+                server.root.output_layout.getBox(view.output.wlr_output, &output_layout_box);
+                const cursor_lx = self.wlr_cursor.x - @intToFloat(f64, output_layout_box.x);
```

Comment:
The result of this is not in layout coordinates but in output-relative coordinates.
```suggestion
                const cursor_ox = self.wlr_cursor.x - @intToFloat(f64, output_layout_box.x);
```